lib/excon/utils.rb

```diff
@@ -87,8 +87,9 @@ def parse_query_string(string)
     end
 
     def default_port?(datum)
-      (datum[:scheme].casecmp?('http') && datum[:port] == 80) ||
-        (datum[:scheme].casecmp?('https') && datum[:port] == 443)
+      datum[:port].nil? ||
```

Comment:
```suggestion
      (!datum[:scheme]&.casecmp?('unix') && datum[:port].nil?)
```

Besides the http/https cases, we also need to account for the `unix` scheme. This check should always return false for `unix`, because there isn't a default there. So to account for that, I think we need to add this extra guard that says if the schema is `unix` we want to return false, regardless of the port value. This appears to get my tests to green locally, fwiw. Make sense?

Reply:
Oh, I see. Changed